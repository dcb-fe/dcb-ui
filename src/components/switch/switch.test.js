import Switch from './switch.vue';
import { shallowMount } from '@vue/test-utils';

describe('Switch', () => {
  describe('视图', () => {
    test('禁用状态', () => {
      const wrapper = shallowMount(Switch, {
        propsData: {
          disabled: true,
        },
      });

      expect(wrapper).toMatchSnapshot();
    });

    test('自定义颜色', () => {
      const wrapper = shallowMount(Switch, {
        propsData: {
          activeBgColor: '#9ABB82',
          inActiveBgColor: '#EAE7E2',
          btnColor: '#CBDDDD',
        },
      });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('逻辑', () => {
    test('开关打开时触发', () => {
      const wrapper = shallowMount(Switch, {
        propsData: {
          active: false,
        },
      });

      wrapper.trigger('click');
    });

    test('开关关闭时触发', () => {
      const wrapper = shallowMount(Switch, {
        propsData: {
          active: false,
        },
      });

      wrapper.trigger('click');
    });
  });
});
