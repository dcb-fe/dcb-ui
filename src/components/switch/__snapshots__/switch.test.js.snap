// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch 视图 禁用状态 1`] = `
<div class="d-switch" style="width: 46px; height: 24px; background: rgb(204, 204, 204); border-color: #cccccc;">
  <div class="d-switch-switch_btn" style="width: 24px; height: 24px; left: 0px; background: rgb(255, 255, 255);"></div>
</div>
`;

exports[`Switch 视图 自定义颜色 1`] = `
<div class="d-switch" style="width: 46px; height: 24px; background: rgb(234, 231, 226); border-color: #eae7e2;">
  <div class="d-switch-switch_btn" style="width: 24px; height: 24px; left: 0px; background: rgb(203, 221, 221);"></div>
</div>
`;
